android/src/main/java/de/patwoz/rn/bluetoothstatemanager/RNBluetoothStateManagerModule.java

```diff
@@ -0,0 +1,257 @@
+
+  private final static int REQUEST_ENABLE_BT = 795;
+  private Promise requestToEnablePromise;
+
+  private final Intent INTENT_OPEN_BLUETOOTH_SETTINGS = new Intent(
+          android.provider.Settings.ACTION_BLUETOOTH_SETTINGS
+  );
+  private final Intent INTENT_REQUEST_ENABLE_BLUETOOTH = new Intent(
+          BluetoothAdapter.ACTION_REQUEST_ENABLE
+  );
+
+  private final static String EVENT_BLUETOOTH_STATE_CHANGE = "EVENT_BLUETOOTH_STATE_CHANGE";
+
+  public RNBluetoothStateManagerModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+    this.reactContext = reactContext;
+
+    if (BluetoothUtils.isBluetoothSupported()) {
+      this.startListenForBluetoothStateChange();
+    }
+  }
+
+  private void destroy() {
+    if (BluetoothUtils.isBluetoothSupported()) {
+      this.stopListenForBluetoothStateChange();
+    }
+    this.removeRequestToEnableListener();
+  }
+
+  @Override
+  public String getName() {
+    return "RNBluetoothStateManager";
+  }
+
+  @Override
+  public Map<String, Object> getConstants() {
+    final Map<String, Object> constants = new HashMap<>();
+    constants.put("EVENT_BLUETOOTH_STATE_CHANGE", EVENT_BLUETOOTH_STATE_CHANGE);
+    return constants;
+  }
+
+  @Override
+  public void onCatalystInstanceDestroy() {
+    super.onCatalystInstanceDestroy();
+    this.destroy();
+  }
+
+  // --------------------------------------------------------------------------------------------- -
+  // BLUETOOTH STATE
+
+  @ReactMethod
+  public void getState(Promise promise) {
+    if (this.handleIfBluetoothNotSupported(promise, false)) {
+      promise.resolve(Constants.BluetoothState.UNSUPPORTED);
+      return;
+    }
+
+    BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
+    promise.resolve(BridgeUtils.fromBluetoothState(bluetoothAdapter.getState()));
+  }
+
+  // --------------------------------------------------------------------------------------------- -
+  // PROGRAMMATICALLY CHANGE BLUETOOTH STATE (not recommended)
+
+  @SuppressLint("MissingPermission")
+  private void setBluetooth(boolean enable) {
+    BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
+    if (enable) {
+      bluetoothAdapter.enable();
+    } else {
+      bluetoothAdapter.disable();
+    }
+  }
+
+  @ReactMethod
+  public void enable(Promise promise) {
+    if (this.handleIfBluetoothNotSupported(promise)) { return; }
+    Activity currentActivity = this.handleCurrentActivity(promise);
+    if (currentActivity == null) { return; }
+
+    if (BluetoothUtils.hasBluetoothAdminPermission(currentActivity)) {
+      this.setBluetooth(true);
+      promise.resolve(null);
+    } else {
+      promise.reject("UNAUTHORIZED", "You are not authorized to do this.");
+    }
+  }
+
+  @ReactMethod
+  public void addListener(String eventName) {}
+
+  @ReactMethod
+  public void removeListeners(Integer count) {}
+
+  @ReactMethod
+  public void disable(Promise promise) {
```

Comment:
looks like the is and enable are basically the same function. maybe we should extract the shared bits

Reply:
Why do we introducing the ebable/disable methods? shouldn't it be just about getting the state?